### PR TITLE
tests: Fix flakiness in Storage retry conformance tests

### PR DESCRIPTION
**Pull request by jskeet**
This works around an issue in gunicorn.

### CODE REVIEW

**Review by amanda-tarafa — APPROVED** [review]
LGTM

**Review by Rishabh-V — APPROVED** [review]
LGTM

File: apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.RetryConformanceTests/RetryConformanceTestFixture.cs [hunk 1]
```diff
@@ -45,6 +45,15 @@ public RetryConformanceTestFixture()
         };
 
         Client = clientBuilder.Build();
+
+        // Workaround for gunicorn broken behavior: if we request keep-alive,
+        // it sends responses that claim the connection will be kept alive,
+        // but then closes the connection anyway. That leads to HttpClient
+        // making a second request on the connection (presumably having not
+        // detected the close) which then fails.
+        HttpClient.DefaultRequestHeaders.ConnectionClose = true;
```
[the remark this answers is not in the document]
Comment (jskeet) [reply apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.RetryConformanceTests/RetryConformanceTestFixture.cs:54]: Yes. One client is used for communicating with the testbench itself; the other is for making "normal" Storage requests (still to the testbench of course).

[the remark this answers is not in the document]
Comment (Rishabh-V) [reply apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.RetryConformanceTests/RetryConformanceTestFixture.cs:54]: Right. I thought we had got rid of the `HttpClient` after the gZip issue was fixed. Thanks.

**Pull request merged**